### PR TITLE
Reorganized Datastore-related papers in the Distributed Systems group

### PR DESCRIPTION
**Pull request by NewAlexandria**
Per #260, this is my suggestion for how to provide more organization to a large topic like Distributed Systems

* There were many papers in the `distributed_systems` folder that related to datastores.  I moved them into the `datastores` folder, and
* updated the `README`s, including links to the out-of-folder papers

### CODE REVIEW

**Comment by jeremyheiler** [comment]
Thanks @NewAlexandria!

@papers-we-love/collaborators, what do you think? In this example, papers in the "datastores" category are also listed in the "distributed systems" category under a "datastores" sub-heading. 

My initial thought is that just linking to the "datastores" category would be easier to maintain.

**Comment by DarrenN** [comment]
@jeremyheiler / @NewAlexandria - any updates?

File: distributed_systems/README.md [hunk 1]
```diff
@@ -228,15 +166,91 @@ By solving the [asymmetries](http://en.wikipedia.org/wiki/Moving_magnet_and_cond
...
+
+* :scroll: [TAO: Facebooks Distributed Data Store for the Social Graph'](../datastores/tao-facebook-distributed-datastore.pdf)
+
+* :scroll: [Transactional storage for geo-replicated systems](../datastores/transactional-storage-for-geo-replicated-systems.pdf)
+
+* :scroll: [Warp: Multi-Key Transactions for Key-Value Stores](../datastores/warp-multi-key-transactions-for-key-value-stores.pdf)
+
+
+### Physics
+
```
Comment (NewAlexandria) [inline distributed_systems/README.md]: @jeremyheiler I did leave the papers in the datastores directory and link to them.

If we linked to the heading, then those distributed-systems-related Datastore papers in the datastore category would not be differentiated from non-dist-sys-datastore papers.  I'm proposing that subcategorization is beneficial (if not necessary) somewhere.

**Comment by DarrenN** [comment]
Can we get this up to speed with master and @jeremyheiler can you provide feedback

File: distributed_systems/README.md (hunk 1 above)
Comment (jeremyheiler) [inline distributed_systems/README.md]: :+1:

**Comment by jeremyheiler** [comment]
I'll merge this when the branch is updated with master.

**Comment by DarrenN** [comment]
@NewAlexandria we're happy to merge once its up to speed with master.

**Comment by NewAlexandria** [comment]
Sure, I'll do that tonight.  Maybe the other one, too

**Comment by NewAlexandria** [comment]
@DarrenN @jeremyheiler resolved

**Comment by DarrenN** [comment]
Many thanks!

**Pull request merged**